application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/Transfer.java

```diff
@@ -68,6 +76,10 @@ public List<Edge> getEdges() {
     return edges;
   }
 
+  public ImmutableSet<StreetMode> getModes() {
```

Comment:
You don't actually need to expose the collection at all. Simply add a method `allowsMode(mode)`. So you don't even have to think about if it's mutable or immutable.

Reply:
I changed `getModes` to `allowsMode`